src/libertem/executor/pipelined.py

```diff
@@ -564,6 +570,25 @@ def _make_spec(
             has_cupy=False,
         ))
         worker_idx += 1
+
+    if isinstance(cudas, int) or len(cudas):
+        # Needed to know if we can assign CUDA workers
+        from libertem.utils.devices import detect
+        avail_cudas = detect()['cudas']
+        if not avail_cudas and cudas:  # needed in case cudas == 0
+            warnings.warn('Specifying CUDA workers on system with '
+                          'no visible CUDA devices',
+                          RuntimeWarning)
+            # If we are assigning from int, just use increasing
+            # device indices even if they are unavailable
+            avail_cudas = itertools.count()
+
+        if isinstance(cudas, int):
+            # Round-Robin-assign to available CUDA devices
+            # Can override by specifying cudas as an iterable
+            cudas_iter = itertools.cycle(avail_cudas)
+            cudas = tuple(next(cudas_iter) for _ in range(cudas))
```

Comment:
Can we extract this as a helper function, and not duplicate it in the two executors?

Reply:
I put this into `executor.utils.__init__` and called it from both versions of `make_spec` (perhaps we could merge the two at some point?).